### PR TITLE
Updated Create AppsV1Deployment resource lifecycle test - +6 endpoint coverage

### LINKED ISSUE #90916: Write AppsV1Deployment resource lifecycle test - +6 endpoint coverage
# Progress `[6/7]`
* [x] Original approval issue: #89340
* [x]  Test approval issue: [kubernetes/kubernetes#90916](https://github.com/kubernetes/kubernetes/issues/90916)
* [x]  Test pr: #92589
* [x]  Two weeks soak start date: 5 Nov 2020 [testgrid-link](https://testgrid.k8s.io/sig-apps#gce&include-filter-by-regex=Deployment&include-filter-by-regex=should%20run%20the%20lifecycle%20of%20a%20Deployment)
* [x]  Code update to remove single flake PR [#96384](https://github.com/kubernetes/kubernetes/pull/96384)
* [x]  Two weeks soak end date: 19 November 2020
* [x]  Test promotion pr: [96487](https://github.com/kubernetes/kubernetes/pull/96487)

This issue is created to allow edit by @Riaankl

/wip
/hold

# Identifying an untested feature Using APISnoop
According to this APIsnoop query, there are still some remaining Deployment endpoints which are untested.

```
SELECT
  operation_id,
  -- k8s_action,
  -- path,
  -- description,
  kind
  FROM untested_stable_endpoints
  where kind like 'Deployment'
  -- WHERE operation_id ilike '%Deployment%'
 ORDER BY kind,operation_id desc
 LIMIT 25
       ;
```

```
                operation_id                |    kind    
--------------------------------------------+------------
 replaceAppsV1NamespacedDeploymentStatus    | Deployment
 readAppsV1NamespacedDeploymentStatus       | Deployment
 patchAppsV1NamespacedDeploymentStatus      | Deployment
 patchAppsV1NamespacedDeployment            | Deployment
 listAppsV1DeploymentForAllNamespaces       | Deployment
 deleteAppsV1CollectionNamespacedDeployment | Deployment
(6 rows)
```

# API Reference and feature documentation
* [Kubernetes API Reference Docs](https://kubernetes.io/docs/reference/kubernetes-api/)
* [client-go - Deployment](https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/Deployment.go)

# The mock test
## Test outline
1. Create a Deployment with a static label
2. Patch the Deployment with a new Label and updated data
3. Get the Deployment to ensure it's patched
4. List all Deployments in all Namespaces find the Deployment(1) ensure that the Deployment is found and is patched
5. Delete Namespaced Deployment(1) via a Collection with a LabelSelector

## Test the functionality in Go
```go
package main

import (
  "encoding/json"
  "fmt"
  "flag"
  "os"
  v1 "k8s.io/api/core/v1"
  appsv1 "k8s.io/api/apps/v1"
  "k8s.io/client-go/dynamic"
  "k8s.io/apimachinery/pkg/runtime"
  "k8s.io/apimachinery/pkg/runtime/schema"
  //"k8s.io/apimachinery/pkg/conversion/unstructured"
  unstructuredv1 "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
  metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
  "k8s.io/client-go/kubernetes"
  "k8s.io/apimachinery/pkg/types"
  "k8s.io/client-go/tools/clientcmd"
  watch "k8s.io/apimachinery/pkg/watch"
)

func main() {
  // uses the current context in kubeconfig
  kubeconfig := flag.String("kubeconfig", fmt.Sprintf("%v/%v/%v", os.Getenv("HOME"), ".kube", "config"), "(optional) absolute path to the kubeconfig file")
  flag.Parse()
  config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
  if err != nil {
      fmt.Println(err)
      return
  }
  // make our work easier to find in the audit_event queries
  config.UserAgent = "live-test-writing"
  // creates the clientset
  ClientSet, _ := kubernetes.NewForConfig(config)
  DynamicClientSet, _ := dynamic.NewForConfig(config)
  deploymentResource := schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "deployments"}

  testDeploymentName := "test-deployment"
  testDeploymentInitialImage := "nginx"
  testDeploymentPatchImage := "alpine"
  testDeploymentUpdateImage := "httpd"
  testDeploymentDefaultReplicas := int32(3)
  testDeploymentMinimumReplicas := int32(1)
  testDeploymentNoReplicas := int32(0)
  testDeploymentLabelSelectors := metav1.LabelSelector{
      MatchLabels: map[string]string{"app": "test-deployment"},
  }
  testNamespaceName := "default"

  fmt.Println("creating a Deployment")
  testDeployment := appsv1.Deployment{
      ObjectMeta: metav1.ObjectMeta{
          Name: testDeploymentName,
          Labels: map[string]string{"test-deployment-static": "true"},
      },
      Spec: appsv1.DeploymentSpec{
          Replicas: &testDeploymentDefaultReplicas,
          Selector: &testDeploymentLabelSelectors,
          Template: v1.PodTemplateSpec{
              ObjectMeta: metav1.ObjectMeta{
                  Labels: testDeploymentLabelSelectors.MatchLabels,
              },
              Spec: v1.PodSpec{
                  Containers: []v1.Container{{
                      Name: testDeploymentName,
                      Image: testDeploymentInitialImage,
                  }},
              },
          },
      },
  }
  _, err = ClientSet.AppsV1().Deployments(testNamespaceName).Create(&testDeployment)
  if err != nil {
      fmt.Println(err)
      return
  }

  fmt.Println("watching for the Deployment to be added")
  dplmtWatchTimeoutSeconds := int64(180)
  dplmtWatch, err := ClientSet.AppsV1().Deployments(testNamespaceName).Watch(metav1.ListOptions{LabelSelector: "test-deployment-static=true", TimeoutSeconds: &dplmtWatchTimeoutSeconds})
  if err != nil {
      fmt.Println(err, "Failed to setup watch on newly created Deployment")
      return
  }

  dplmtWatchChan := dplmtWatch.ResultChan()
  for event := range dplmtWatchChan {
      if event.Type == watch.Added {
          break
      }
  }
  defer func() {
    fmt.Println("deleting the Deployment")
    err = ClientSet.AppsV1().Deployments(testNamespaceName).DeleteCollection(&metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "test-deployment-static=true"})
    if err != nil {
      fmt.Println(err)
      return
    }
    for event := range dplmtWatchChan {
      deployment, ok := event.Object.(*appsv1.Deployment)
      if ok != true {
        fmt.Println("unable to convert event.Object type")
        return
      }
      if event.Type == watch.Deleted && deployment.ObjectMeta.Name == testDeploymentName {
        break
      }
    }
  }()
  fmt.Println("waiting for all Replicas to be Ready")
  for event := range dplmtWatchChan {
      deployment, ok := event.Object.(*appsv1.Deployment)
      if ok != true {
          fmt.Println("unable to convert event.Object type")
          return
      }
      if deployment.Status.AvailableReplicas == testDeploymentDefaultReplicas &&
         deployment.Status.ReadyReplicas == testDeploymentDefaultReplicas {
          break
      }
  }

  fmt.Println("patching the Deployment")
  deploymentPatch, err := json.Marshal(map[string]interface{}{
      "metadata": map[string]interface{}{
          "labels": map[string]string{"test-deployment": "patched"},
      },
      "spec": map[string]interface{}{
          "replicas": testDeploymentMinimumReplicas,
          "template": map[string]interface{}{
              "spec": map[string]interface{}{
                  "containers": []map[string]interface{}{{
                      "name": testDeploymentName,
                      "image": testDeploymentPatchImage,
                      "command": []string{"/bin/sleep", "100000"},
                  }},
              },
          },
      },
  })
  if err != nil {
      fmt.Println(err, "failed to Marshal Deployment JSON patch")
      return
  }
  _, err = ClientSet.AppsV1().Deployments(testNamespaceName).Patch(testDeploymentName, types.StrategicMergePatchType, []byte(deploymentPatch))
  if err != nil {
       fmt.Println(err, "failed to patch Deployment")
       return
  }

  for event := range dplmtWatchChan {
      if event.Type == watch.Modified {
          break
      }
  }
  fmt.Println("waiting for Replicas to scale")
  for event := range dplmtWatchChan {
      deployment, ok := event.Object.(*appsv1.Deployment)
      if ok != true {
          fmt.Println("unable to convert event.Object type")
          return
      }
      if deployment.Status.AvailableReplicas == testDeploymentMinimumReplicas &&
         deployment.Status.ReadyReplicas == testDeploymentMinimumReplicas {
          break
      }
  }


  fmt.Println("listing Deployments")
  deploymentsList, err := ClientSet.AppsV1().Deployments("").List(metav1.ListOptions{LabelSelector: "test-deployment-static=true"})
  if err != nil {
      fmt.Println(err, "failed to list Deployments")
      return
  }
  foundDeployment := false
  for _, deploymentItem := range deploymentsList.Items {
      if deploymentItem.ObjectMeta.Name == testDeploymentName &&
         deploymentItem.ObjectMeta.Namespace == testNamespaceName &&
         deploymentItem.ObjectMeta.Labels["test-deployment-static"] == "true" &&
         *deploymentItem.Spec.Replicas == testDeploymentMinimumReplicas &&
         deploymentItem.Spec.Template.Spec.Containers[0].Image == testDeploymentPatchImage {
          foundDeployment = true
          break
      }
  }
  if foundDeployment != true {
      fmt.Println("unable to find the Deployment in list")
      return
  }

  fmt.Println("updating the DeploymentStatus")
  testDeploymentUpdate := testDeployment
  testDeploymentUpdate.ObjectMeta.Labels["test-deployment"] = "updated"
  testDeploymentUpdate.Spec.Template.Spec.Containers[0].Image = testDeploymentUpdateImage
  testDeploymentDefaultReplicasPointer := &testDeploymentDefaultReplicas
  testDeploymentUpdate.Spec.Replicas = testDeploymentDefaultReplicasPointer
  testDeploymentUpdate.Status.ReadyReplicas = testDeploymentNoReplicas
  testDeploymentUpdateUnstructuredMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&testDeploymentUpdate)
  if err != nil {
      fmt.Println(err, "failed to convert to unstructured")
  }
  testDeploymentUpdateUnstructured := unstructuredv1.Unstructured{
      Object: testDeploymentUpdateUnstructuredMap,
  }
	// currently this hasn't been able to hit the endpoint replaceAppsV1NamespacedDeploymentStatus
  _, err = DynamicClientSet.Resource(deploymentResource).Namespace(testNamespaceName).Update(&testDeploymentUpdateUnstructured, metav1.UpdateOptions{})//, "status")
  if err != nil {
      fmt.Println(err, "failed to update the DeploymentStatus")
      return
  }
  for event := range dplmtWatchChan {
      if event.Type == watch.Modified {
          break
      }
  }

  fmt.Println("fetching the DeploymentStatus")
  deploymentGetUnstructured, err := DynamicClientSet.Resource(deploymentResource).Namespace(testNamespaceName).Get(testDeploymentName, metav1.GetOptions{}, "status")
  if err != nil {
      fmt.Println(err, "failed to fetch the Deployment")
      return
  }
  deploymentGet := appsv1.Deployment{}
  err = runtime.DefaultUnstructuredConverter.FromUnstructured(deploymentGetUnstructured.Object, &deploymentGet)
  if err != nil {
      fmt.Println(err, "failed to convert the unstructured response to a Deployment")
      return
  }
  if ! (deploymentGet.Spec.Template.Spec.Containers[0].Image == testDeploymentUpdateImage || deploymentGet.Status.ReadyReplicas == testDeploymentNoReplicas || deploymentGet.ObjectMeta.Labels["test-deployment"] == "updated") {
      fmt.Println("failed to update the Deployment (did not return correct values)")
      return
  }
  for event := range dplmtWatchChan {
      if event.Type == watch.Modified {
          break
      }
  }
  for event := range dplmtWatchChan {
      deployment, ok := event.Object.(*appsv1.Deployment)
      if ok != true {
          fmt.Println("failed to convert event Object to a Deployment")
          return
      }
      if deployment.Status.ReadyReplicas == testDeploymentDefaultReplicas {
          break
      }
  }

  fmt.Println("patching the DeploymentStatus")
  deploymentStatusPatch, err := json.Marshal(map[string]interface{}{
      "metadata": map[string]interface{}{
          "labels": map[string]string{"test-deployment": "patched-status"},
      },
      "status": map[string]interface{}{
          "readyReplicas": testDeploymentNoReplicas,
      },
  })
  if err != nil {
      fmt.Println(err, "failed to Marshal Deployment JSON patch")
      return
  }
  DynamicClientSet.Resource(deploymentResource).Namespace(testNamespaceName).Patch(testDeploymentName, types.StrategicMergePatchType, []byte(deploymentStatusPatch), metav1.PatchOptions{}, "status")

  fmt.Println("fetching the DeploymentStatus")
  deploymentGetUnstructured, err = DynamicClientSet.Resource(deploymentResource).Namespace(testNamespaceName).Get(testDeploymentName, metav1.GetOptions{}, "status")
  if err != nil {
      fmt.Println(err, "failed to fetch the DeploymentStatus")
      return
  }
  deploymentGet = appsv1.Deployment{}
  err = runtime.DefaultUnstructuredConverter.FromUnstructured(deploymentGetUnstructured.Object, &deploymentGet)
  if err != nil {
      fmt.Println(err, "failed to convert the unstructured response to a Deployment")
      return
  }
  if ! (deploymentGet.Spec.Template.Spec.Containers[0].Image == testDeploymentUpdateImage || deploymentGet.Status.ReadyReplicas == 0 || deploymentGet.ObjectMeta.Labels["test-deployment"] == "patched-status") {
      fmt.Println("failed to update the Deployment (did not return correct values)")
      return
  }
  for event := range dplmtWatchChan {
      if event.Type == watch.Modified {
          break
      }
  }
  for event := range dplmtWatchChan {
      deployment, ok := event.Object.(*appsv1.Deployment)
      if ok != true {
          fmt.Println("failed to convert event Object to a Deployment")
          return
      }
      if deployment.Status.ReadyReplicas == testDeploymentDefaultReplicas {
          break
      }
  }

  // write test here
  fmt.Println("[status] complete")
}
```

```go
creating a Deployment
watching for the Deployment to be added
waiting for all Replicas to be Ready
patching the Deployment
waiting for Replicas to scale
listing Deployments
updating the DeploymentStatus
fetching the DeploymentStatus
patching the DeploymentStatus
fetching the DeploymentStatus
[status] complete
deleting the Deployment
```

# Verifying increase it coverage with APISnoop
Discover useragents:

```
select distinct useragent from audit_event where bucket='apisnoop' and useragent not like 'kube%' and useragent not like 'coredns%' and useragent not like 'kindnetd%' and useragent like 'live%';
```

```
     useragent     
-------------------
 live-test-writing
(1 row)
```

List endpoints hit by the test:

```
select * from endpoints_hit_by_new_test where useragent like 'live%'; 
```

```
     useragent     |                operation_id                | hit_by_ete | hit_by_new_test 
-------------------+--------------------------------------------+------------+-----------------
 live-test-writing | createAppsV1NamespacedDeployment           | t          |               1
 live-test-writing | deleteAppsV1CollectionNamespacedDeployment | f          |               1
 live-test-writing | listAppsV1DeploymentForAllNamespaces       | f          |               1
 live-test-writing | listAppsV1NamespacedDeployment             | t          |               1
 live-test-writing | patchAppsV1NamespacedDeployment            | f          |               1
 live-test-writing | patchAppsV1NamespacedDeploymentStatus      | f          |               1
 live-test-writing | readAppsV1NamespacedDeploymentStatus       | f          |               2
 live-test-writing | replaceAppsV1NamespacedDeployment          | t          |               1
(8 rows)
```

Display endpoint coverage change:

```
select * from projected_change_in_coverage;
```

```
   category    | total_endpoints | old_coverage | new_coverage | change_in_number 
---------------+-----------------+--------------+--------------+------------------
 test_coverage |             445 |          181 |          186 |                5
(1 row)
```

# Final notes
If a test with these calls gets merged, ****test coverage will go up by 5 points****

This test is also created with the goal of conformance promotion.

### PR DESCRIPTION
**Pull request by riaankleinhans**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it:**
This PR adds a test to test the following untested endpoints:
deleteAppsV1CollectionNamespacedDeployment
listAppsV1DeploymentForAllNamespaces
patchAppsV1NamespacedDeployment
patchAppsV1NamespacedDeploymentStatus
readAppsV1NamespacedDeploymentStatus
replaceAppsV1NamespacedDeploymentStatus

**Which issue(s) this PR fixes:**
Fixes #90916

**Special notes for your reviewer:**
Adds +5 endpoint test coverage (good for conformance)

**Does this PR introduce a user-facing change?:**
```
NONE
```

**Release note:**
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:**
```
NONE
```

/sig testing
/sig architecture
/area conformance

### CODE REVIEW

**Comment by liggitt** [comment]
see flakes in kind runs:
* https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93458/pull-kubernetes-e2e-kind/1288909664666259456
* https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93458/pull-kubernetes-e2e-kind/1288899169599098881
* https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/93458/pull-kubernetes-e2e-kind-ipv6/1288245896345030657

was anything modified from the version reverted in https://github.com/kubernetes/kubernetes/pull/93405?

File: test/e2e/apps/deployment.go [hunk 1]
```diff
@@ -134,6 +142,300 @@ var _ = SIGDescribe("Deployment", func() {
 	})
 	// TODO: add tests that cover deployment.Spec.MinReadySeconds once we solved clock-skew issues
 	// See https://github.com/kubernetes/kubernetes/issues/29229
+
+	ginkgo.It("should run the lifecycle of a Deployment", func() {
```
Comment (oomichi) [inline test/e2e/apps/deployment.go:146]: Can we separate this into multiple tests?
Current test is one of big tests by comparing the other tests in this test package and that would make it difficult to investigate issues if they happen.

**Review by oomichi — COMMENTED** [review]
/cc @oomichi

File: test/e2e/apps/deployment.go (hunk 1 above)
Comment (liggitt) [reply test/e2e/apps/deployment.go:146]: every individual test has setup/tear down cost, and our e2e jobs already trend long. A single coherent test that walks an object through its paces seems better to me.

**Comment by heyste** [comment]
Checking how the test runs/performs as local testing hasn't found any issues with the test to date

/hold

**Comment by heyste** [comment]
pull-kubernetes-e2e-gce-ubuntu-containerd: multiple test failures for other tests.

/test pull-kubernetes-e2e-gce-ubuntu-containerd

**Comment by heyste** [comment]
pull-kubernetes-verify: "Process did not finish before 2h0m0s timeout"

/test pull-kubernetes-verify

**Comment by heyste** [comment]
Before the new [deflake guide](https://gist.github.com/liggitt/6a3a2217fa5f846b52519acfc0ffece0#deflaking-e2e-tests) was available a lot of time was taken to review the code and check how it worked with a low resource cluster (on a stressed VM) with the goal of checking that it was reliable as possible. I wasn't able to get this test to fail once.

In July there where a lot of test infrastructure issues that created a lot of noise that may of masked an unknown issue with the test. The deflake guide has some great advice that will be used for any future issues.

The goal is to get an updated set of logs that show that this PR is solid before asking for a merge request.

**Comment by heyste** [comment]
/test pull-kubernetes-e2e-gce
flake
```
Kubernetes e2e suite: [sig-cli] Kubectl client Kubectl server-side dry-run should check if kubectl can dry-run update Pods [Conformance] 
test/e2e/framework/framework.go:629 Oct 27 20:01:57.892: Unexpected error:
...
Error from server (Conflict): error when replacing \"STDIN\": Operation cannot be fulfilled on pods \"e2e-test-httpd-pod\": the object has been modified; please apply your changes to the latest version and try again\n\nerror:\nexit status 1
```

**Comment by heyste** [comment]
/test pull-kubernetes-e2e-gce-ubuntu-containerd
flake #95832
```
Kubernetes  e2e suite: [sig-cli] Kubectl client Kubectl server-side dry-run should  check if kubectl can dry-run update Pods [Conformance] 
 Error from server (Conflict): error when replacing "STDIN": Operation cannot be fulfilled on pods "e2e-test-httpd-pod": the object has been modified; please apply your changes to the latest version and try again
```

**Comment by heyste** [comment]
/test pull-kubernetes-e2e-gce-ubuntu-containerd
flake

```
Kubernetes  e2e suite: [sig-storage] In-tree Volumes [Driver:  local][LocalVolumeType: dir-link-bindmounted] [Testpattern:  Pre-provisioned PV (default fs)] subPath should support non-existent  path
test/e2e/storage/testsuites/subpath.go:191
...
Terminated:&ContainerStateTerminated{ExitCode:128,Signal:0,Reason:StartError,Message:failed to create containerd task: failed to start io pipe copy: unable to copy pipes: containerd-shim: opening w/o fifo \"/run/containerd/io.containerd.grpc.v1.cri/containers/c0dc3a7db1198a4ea08ff211884e71000b202d59d49a38a30c0116f5b5f2bcaa/io/249649854/c0dc3a7db1198a4ea08ff211884e71000b202d59d49a38a30c0116f5b5f2bcaa-stdout\"
```

**Comment by Riaankl** [comment]
Thank you @smarterclayton 
/lgtm

**Comment by heyste** [comment]
/test pull-kubernetes-bazel-build
flakes
```
go_test: //staging/src/k8s.io/apiserver/pkg/server/filters/go_default_test:run_1_of_2 expand_less 	0s
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //staging/src/k8s.io/apiserver/pkg/server/filters:go_default_test
-----------------------------------------------------------------------------
2020/11/04 00:06:13 httptest.Server blocked in Close after 5 seconds, waiting for connections:
  *net.TCPConn 0xc0003f2068 127.0.0.1:46596 in state active
  *net.TCPConn 0xc0003f2060 127.0.0.1:46594 in state active
Terminated
-- Test timed out at 2020-11-04 00:11:06 UTC --
go_test: //staging/src/k8s.io/apiserver/pkg/server/filters/go_default_test:run_2_of_2 expand_less 	0s
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //staging/src/k8s.io/apiserver/pkg/server/filters:go_default_test
-----------------------------------------------------------------------------
2020/11/04 00:06:13 httptest.Server blocked in Close after 5 seconds, waiting for connections:
  *net.TCPConn 0xc000186050 127.0.0.1:59230 in state active
  *net.TCPConn 0xc000186040 127.0.0.1:59226 in state active
Terminated
-- Test timed out at 2020-11-04 00:11:06 UTC --
```

**Comment by heyste** [comment]
/test pull-kubernetes-bazel-test
hit wrong test

**Comment by heyste** [comment]
/test pull-kubernetes-bazel-test
flake

```
//staging/src/k8s.io/apiserver/pkg/server/filters:go_default_test       TIMEOUT in 1 out of 2 in 315.0s
  Stats over 2 runs: max = 315.0s, min = 2.1s, avg = 158.5s, dev = 156.5s
  /bazel-scratch/.cache/bazel/_bazel_root/7989b31489f31aee54f32688da2f0120/execroot/io_k8s_kubernetes/bazel-out/k8-fastbuild/testlogs/staging/src/k8s.io/apiserver/pkg/server/filters/go_default_test/run_1_of_2/test.log
//staging/src/k8s.io/legacy-cloud-providers/vsphere:go_default_test      FAILED in 1 out of 2 in 3.2s
  Stats over 2 runs: max = 3.2s, min = 3.1s, avg = 3.1s, dev = 0.1s
  /bazel-scratch/.cache/bazel/_bazel_root/7989b31489f31aee54f32688da2f0120/execroot/io_k8s_kubernetes/bazel-out/k8-fastbuild/testlogs/staging/src/k8s.io/legacy-cloud-providers/vsphere/go_default_test/run_1_of_2/test.log
```

**Comment by Riaankl** [comment]
[Testgrid](https://testgrid.k8s.io/sig-apps#gce&include-filter-by-regex=Deployment&include-filter-by-regex=should%20run%20the%20lifecycle%20of%20a%20Deployment)

**Comment by Riaankl** [comment]
Single flaky test run [7 Nov ](https://testgrid.k8s.io/sig-apps#gce&include-filter-by-regex=Deployment&include-filter-by-regex=should%20run%20the%20lifecycle%20of%20a%20Deployment&width=20) fixed by #96384

**Pull request merged**